domain/relation/state/relation_test.go

```diff
@@ -346,6 +349,144 @@ func (s *relationSuite) TestGetRelationEndpointsRelationNotFound(c *gc.C) {
 	c.Assert(err, jc.ErrorIs, relationerrors.RelationNotFound)
 }
 
+func (s *relationSuite) TestGetRelationsStatusForUnit(c *gc.C) {
+	// Arrange: Add a relation with two endpoints.
+	relationUUID := "fake-relation-uuid"
+	charmRelationUUID1 := "fake-charm-relation-uuid-1"
+	applicationEndpointUUID1 := "fake-application-endpoint-uuid-1"
+	relationEndpointUUID1 := "fake-relation-endpoint-uuid-1"
+	endpoint1 := internalrelation.Endpoint{
+		ApplicationName: s.constants.fakeApplicationName1,
+		Relation: internalcharm.Relation{
+			Name:  "fake-endpoint-name-1",
+			Role:  internalcharm.RoleProvider,
+			Scope: internalcharm.ScopeGlobal,
+		},
+	}
+	charmRelationUUID2 := "fake-charm-relation-uuid-2"
+	applicationEndpointUUID2 := "fake-application-endpoint-uuid-2"
+	relationEndpointUUID2 := "fake-relation-endpoint-uuid-2"
+	endpoint2 := internalrelation.Endpoint{
+		ApplicationName: s.constants.fakeApplicationName2,
+		Relation: internalcharm.Relation{
+			Name:  "fake-endpoint-name-2",
+			Role:  internalcharm.RoleRequirer,
+			Scope: internalcharm.ScopeGlobal,
+		},
+	}
+	s.addCharmRelation(c, fakeCharmUUID1, charmRelationUUID1, endpoint1.Relation)
+	s.addCharmRelation(c, fakeCharmUUID2, charmRelationUUID2, endpoint2.Relation)
+	s.addApplicationEndpoint(c, applicationEndpointUUID1, s.constants.fakeApplicationUUID1, charmRelationUUID1)
+	s.addApplicationEndpoint(c, applicationEndpointUUID2, s.constants.fakeApplicationUUID2, charmRelationUUID2)
+	s.addRelation(c, relationUUID)
+	s.addRelationEndpoint(c, relationEndpointUUID1, relationUUID, applicationEndpointUUID1)
+	s.addRelationEndpoint(c, relationEndpointUUID2, relationUUID, applicationEndpointUUID2)
+
+	// Arrange: Add a unit.
+	unitUUID := "fake-unit-uuid"
+	s.addUnit(c, unitUUID, "unit-name", s.constants.fakeApplicationUUID1)
+
+	// Arrange: Add unit to relation and set relation status.
+	relUnitUUID := "fake-relation-unit-uuid"
+	s.addRelationUnit(c, unitUUID, relationUUID, relUnitUUID, true)
+	s.addRelationStatus(c, relationUUID, corestatus.Suspended)
+
+	expectedResults := []relation.RelationUnitStatusResult{{
+		Endpoints: []internalrelation.Endpoint{endpoint1, endpoint2},
+		InScope:   true,
+		Suspended: true,
+	}}
+
+	// Act: Get relation status for unit.
+	results, err := s.state.GetRelationsStatusForUnit(context.Background(), coreunit.UUID(unitUUID))
+
+	// Assert:
+	c.Assert(err, jc.ErrorIsNil, gc.Commentf("(Assert): %v",
+		errors.ErrorStack(err)))
+	c.Check(results, gc.DeepEquals, expectedResults)
+}
+
+// TestGetRelationsStatusForUnit checks that GetRelationStatusesForUnit works
+// well with peer relations.
+func (s *relationSuite) TestGetRelationsStatusForUnitPeer(c *gc.C) {
+	// Arrange: Add two peer relations with one endpoint each.
+	relationUUID1 := "fake-relation-uuid-1"
+	charmRelationUUID1 := "fake-charm-relation-uuid-1"
+	applicationEndpointUUID1 := "fake-application-endpoint-uuid-1"
+	relationEndpointUUID1 := "fake-relation-endpoint-uuid-1"
```

Comment:
use `GenRelationUUID(c)` etc.

Reply:
I've switched to using them for entities that have them here. Though I also [commented here](https://github.com/juju/juju/pull/19294#discussion_r2012756557) about how this form can be useful for testing.